rfcs/react-components/components/dev-docs.md

```diff
@@ -0,0 +1,66 @@
+As a new developer to the Fluent repo, but long time Fluent consumer and fan, the supporting documentation has been a rough experience. This has forced me to randomize other members of the team which slows us both down. This RFC seeks to lay out a plan to modernize the Fluent dev docs.
+
+## Current issues:
+
+- Mixture of v6, 7, 8 and 9 specific docs.
+- Docs seemed to have stopped being updated in the early days of v9 development.
+- Mixture of process, set up, and reference docs.
+- RFC’s seem to be a large part of the Fluent process now, but there’s no clear way for an outsider to observe or contribute to them.
+- Use of Github’s wiki isn't the best technical solution.
+
+## Goals of the new system:
+
+- Be more accessible to developers in their work flow.
+- Help developers onboard faster without distracting core team members.
+- Provide technical documentation and guides for common patterns and practices that do not make sense to live in examples or code samples.
+- Focus attention on areas under active development (web and react components)
+
+## Non-Goals
+
+- Provide an ongoing communications platform for project statuses, road maps and planning.
```

Comment:
FYI, the charting team has started using docusaurus to host our technical documents. We coudn't find an easy way to publish our guides and dev docs in the existing fluent repo or the fluent docsite. 
https://microsoft.github.io/fluentui-charting-contrib/

Reply:
Would a dedicated directory in the __docs__  help you all?

Reply:
More than the docs folder structuring we need a way for users to be able to easily navigate through the docs, search for specific terms, and interact/contribute with the docs content.

Reply:
Awesome - these are great requirements. Thanks so much for the feedback and contributions Atishay!

Reply:
Following up on this one:

Our goals for this effort are scope to contribution to the repo. We're going to stop short of trying to support outward or process related communication. Contributors are always encouraged to engage with PRs as they see fit.